src/edu/ucsd/library/xdre/tab/SubjectMatching.java

```diff
@@ -0,0 +1,185 @@
+import edu.ucsd.library.xdre.utils.DAMSClient;
+import edu.ucsd.library.xdre.utils.DAMSRepository;
+
+/**
+ * SubjectMatching: pre-test for matching subjects
+ * 
+ * @author lsitu
+ * @since 2018-01-04
+**/
+public class SubjectMatching {
+	private static Logger log = Logger.getLogger(RDFDAMS4ImportTsHandler.class);
+	private static String[] MADS_SUBJECTS = {"PersonalName", "ConferenceName", "CorporateName", "FamilyName", "Topic",
+		"GenreForm", "Geographic", "Occupation", "Temporal"};
+
+	private DAMSClient damsClient = null;
+	private List<File> rdfFiles = null;
+	private List<String> models = null;
+
+	/**
+	 * Constructor for SubjectMatching
+	 * @param fileNames
+	 * @param filesPaths
+	 * @throws Exception
+	 */
+	public SubjectMatching (DAMSClient damsClient, List<File> rdfFiles, List<String> subjectTypes) throws Exception
+	{
+		this.damsClient = damsClient;
+		this.rdfFiles = rdfFiles;
+		this.models = convertToModelName(subjectTypes);
+	}
+
+	/**
+	 * Convert the subject types to model class name
+	 * @param subjectTypes
+	 * @return
+	 */
+	private List<String> convertToModelName(List<String> subjectTypes) {
+		List<String> subjectModels = new ArrayList<>();
+
+		for (String subjectType : subjectTypes) {
+			String subjectName = subjectType.substring(subjectType.lastIndexOf(":") + 1).replace(" name", "Name");
+			subjectName = subjectName.substring(0, 1).toUpperCase() + subjectName.substring(1, subjectName.length());
+			if (subjectName.equals("Genre"))
+				subjectName = "GenreForm";
+
+			if (madsSubject(subjectName))
+				subjectName = "mads:" + subjectName;
+			else
+				subjectName = "dams:" + subjectName;
+
+			subjectModels.add(subjectName);
+		}
+		return subjectModels;
+	}
+
+	/**
+	 * Determine whether a subject is a MADS subject
+	 * @param subjectName
+	 * @return
+	 */
+	private boolean madsSubject(String subjectName) {
+		return Arrays.asList(MADS_SUBJECTS).contains(subjectName);
+	}
+
+	/**
+	 * Method to retrieve match subjects
+	 * @return CSV format string
+	 * @throws IOException 
+	 * @throws LoginException 
+	 */
+	public String getMatchSubjects () throws LoginException, IOException
+	{
+		StringBuilder csvBuilder = new StringBuilder();
+		String header = escapeRow("subject type", "subject term", "result", "ARK", "exactMatch", "closeMatch");
+		csvBuilder.append(header);
+
+		DAMSRepository damsRepository = DAMSRepository.getRepository();
+		for (File rdfFile : rdfFiles) {
+			SAXReader saxReader = new SAXReader();
+			try {
+				boolean result = false;
+				Document srcDoc = saxReader.read(rdfFile);
+				Node labelNode = srcDoc.selectSingleNode("//mads:authoritativeLabel");
+				String subjectTerm  = labelNode.getText();
+				String modelName= getSubjectTypeLabel(labelNode.getParent().getName());
+				
+				for (String model : models) {
+					List<String> matchedSubjects = damsRepository.findAuthority(model, subjectTerm);
+					if (matchedSubjects != null) {
```

Comment:
could we flatten the conditional check here into a guard clause? something like:
```
if (matchedSubjects == null) {
  continue;
}
...
```

Reply:
Yes. we can.